File: services/api/app/schemas/practice.py
from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel

from app.schemas.topic import TopicOut

class PracticeCategorySummary(BaseModel):
    slug: str
    name: str
    description: Optional[str]
    icon: Optional[str]
    total_questions: int
    difficulty: str
    difficulties: List[str]
    quiz_id: Optional[int] = None
    organization_id: Optional[int] = None
    topics: List[TopicOut] = []


class PracticeQuestionOption(BaseModel):
    id: int
    text: str
    is_correct: bool

    model_config = {
        "from_attributes": True,
    }


class PracticeQuestion(BaseModel):
    id: int
    prompt: str
    explanation: Optional[str]
    difficulty: Optional[str]
    options: List[PracticeQuestionOption]

    model_config = {
        "from_attributes": True,
    }


class PracticeCategoryDetail(BaseModel):
    slug: str
    name: str
    description: Optional[str]
    icon: Optional[str]
    total_questions: int
    difficulty: str
    questions: List[PracticeQuestion]
    organization_id: Optional[int] = None
    topics: List[TopicOut] = []

File: app/web/src/types/practice.ts
export interface PracticeCategorySummary {
  slug: string
  name: string
  description: string | null
  icon: string | null
  total_questions: number
  difficulty: string
  difficulties: string[]
  quiz_id?: number | null
  organization_id?: number | null
}

export interface PracticeQuestionOption {
  id: number
  text: string
  is_correct: boolean
}

export interface PracticeQuestion {
  id: number
  prompt: string
  explanation: string | null
  difficulty: string | null
  options: PracticeQuestionOption[]
}

export interface PracticeCategoryDetail {
  slug: string
  name: string
  description: string | null
  icon: string | null
  total_questions: number
  difficulty: string
  questions: PracticeQuestion[]
  organization_id?: number | null
}
